web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -50,43 +48,23 @@ export class LayerDialogComponent implements OnDestroy {
   layer?: Layer;
   layerName!: string;
   projectId?: string;
-  activeProject$: Observable<Project>;
   subscription: Subscription = new Subscription();
   fieldTypes = FieldType;
   fields: List<Field>;
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) data: { layerId: string },
+    @Inject(MAT_DIALOG_DATA) data: { layerId: string; project: Project },
```

Comment:
To minimize the surface area of the API of this component, can we pass in only what's needed, i.e. projectId and Layer?